src/main.rs

```diff
@@ -79,14 +83,14 @@ fn main() -> Result<(), Box<dyn Error>> {
     let matches = match opts.parse(std::env::args()) {
         Ok(it) => it,
         Err(err) => {
-            usage("commit-analzer", opts);
+            usage(opts);
             return Err(err.into());
         }
     };
-    if matches.free.len() < 2 {
-        usage(&matches.free[0], opts);
+    if matches.opt_present("h") || matches.free.len() < 0x2 {
```

Comment:
In C, there is a feature named "Numeral Annotation".  It is considered good style to append the corresponding letters to a number in order to indicate the intended range.  For instance, `123ULL` will then be assumed to be of the type `unsigned long long int` and not just a simple `int`.

The only two numeric primitives which do not have such an annotation are `int` and `double`.  `double` numbers also accept the `int` representation but may behave other than expected then.  In order to distinct the representation of an `int` from the representation of a `double`, I prefer the hexadecimal representation for integers.  Hexadecimal number will always be parsed to `int`s, so the meaning of `0x2A` (42) is always clear to both the compiler and the software engineers reading the code:  `0x2A` is intended to be a simple `int`.

The hexadecimal representation is also wonderful to abbreviate numbers, especially those which are powers to two.  Often, C / C++ headers specify a set of magic numbers using hexadecimal numbers, for instance OpenGL.

I used the hexadecimal representation here in Rust, as well, since I always use it to indicate a number being intended to be compiled as an integer.  More precisely, I specify integers numbers usually as hexadecimals since, due to the C / C++ coding, it became normal to me.

If you dislike it, I can change the representation back to decimal.

Reply:
In rust this is not a problem, since
```rust
let d: f64 = 42;
```
would not compile. You would have to write either `42.0` or `42f64` for that.

Reply:
Done.